lib/lambda-functions/application-handlers/src/permissionSetSync.ts

```diff
@@ -143,35 +132,66 @@ export const handler = async (event: SNSEvent) => {
         await ssoAdminClientObject.send(new ListInstancesCommand({}));
 
       const instanceArn = resolvedInstances.Instances?.[0].InstanceArn + "";
-      const staticSSOPayload: StaticSSOPayload = {
-        InstanceArn: instanceArn,
-        TargetType: "AWS_ACCOUNT",
-        PrincipalType: "GROUP",
-      };
+
+      const identityStoreId = resolvedInstances.Instances?.[0].IdentityStoreId;
 
       await Promise.all(
         relatedLinks.Items.map(async (Item) => {
-          const relatedGroups: QueryCommandOutput =
-            await ddbDocClientObject.send(
-              // QueryCommand is a pagniated call, however the logic requires
-              // checking only if the result set is greater than 0
-              new QueryCommand({
-                TableName: groupsTableName,
-                IndexName: "groupName",
-                KeyConditionExpression: "#groupname = :groupname",
-                ExpressionAttributeNames: { "#groupname": "groupName" },
-                ExpressionAttributeValues: { ":groupname": Item.groupName },
-              })
-            );
-          if (relatedGroups.Items && relatedGroups.Items.length !== 0) {
+          // Compute user/group name based on whether Active Directory is the user store
+          let principalId = "0";
+          let principalNameToLookUp = Item.principalName;
+          if (adUsed === "true" && domainName !== "") {
+            principalNameToLookUp = `${Item.principalName}@${domainName}`;
+          }
+          if (Item.principalType === "GROUP") {
```

Comment:
I got a bit confused reading through this if, is it possible it could be broken out into logical parts / functions?

Reply:
Is the recommendation that the principal ID resolution logic be re-factored into a shared utility function?

Reply:
This is now refactored, all principal ID resolution is refactored into a utility function